cpp/src/table/row_operators.cu

```diff
@@ -264,18 +282,23 @@ using column_checker_fn_t = std::function<void(column_view const&)>;
  */
 void check_lex_compatibility(table_view const& input)
 {
-  // Basically check if there's any LIST hiding anywhere in the table
+  // Basically check if there's any LIST of STRUCT or STRUCT of LIST hiding anywhere in the table
```

Comment:
We can handle STRUCT of LIST, right?
```suggestion
  // Basically check if there's any LIST of STRUCT hiding anywhere in the table
```

Reply:
That's what I thought as well, and in https://github.com/rapidsai/cudf/pull/11129/commits/d58ad8064d2c2812777f1dbb0ba9299dd5c9073d I removed the actual check for this case, but it looks like that caused CI to fail ([and not just because of an `EXPECT_THROW`](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-gpu-test/CUDA=11.0,GPU_LABEL=driver-450,LINUX_VER=centos7,PYTHON=3.8/10269/testReport/junit/(root)/groupby_structs_test_0/lists_are_unsupported/)). I can't reproduce those failures locally, so I am inclined to reinstate the check for now (i.e. revert https://github.com/rapidsai/cudf/pull/11129/commits/d58ad8064d2c2812777f1dbb0ba9299dd5c9073d), merge this PR, then investigate the `STRUCT` of `LIST` case in a follow-up PR. What do you think?

Reply:
For now I've reverted that commit, so STRUCT of LIST is still disallowed. Let me know if you feel strongly that we should resolve the issue in this PR (I'll preface this with the fact that I haven't investigated any further yet except to say that I can't reproduce this locally at the moment, so it could be some deep complex issue or just some silly oversight in the test).

Reply:
Happy to defer to a follow-up PR. See comment below, we should probably file an issue with a task list of follow-up work.

Reply:
#11672